AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m

```diff
@@ -520,6 +519,55 @@ - (void)testPerformCosmosDbAsyncOperationWithHttpClient {
   XCTAssertEqualObjects(expectedURLString, [actualURL absoluteString]);
 }
 
+- (void)testPerformCosmosDbOperationWithPartitionWithErrorResponseWithoutError {
+
+  // If
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Received an error code from Cosmos DB"];
+  __block NSError *actualError;
+  __block NSHTTPURLResponse *actualResponse;
+  MSTokenResult *testToken = [self mockTokenFetchingWithError:nil];
+
+  // Mock CosmosDB requests.
+  OCMStub([self.cosmosDbMock performCosmosDbAsyncOperationWithHttpClient:OCMOCK_ANY
+                                                             tokenResult:testToken
+                                                              documentId:kMSDocumentIdTest
+                                                              httpMethod:kMSHttpMethodPost
+                                                                    body:OCMOCK_ANY
+                                                       additionalHeaders:OCMOCK_ANY
+                                                       completionHandler:OCMOCK_ANY])
+      .andDo(^(NSInvocation *invocation) {
+        MSHttpRequestCompletionHandler cosmosdbOperationCallback;
+        [invocation getArgument:&cosmosdbOperationCallback atIndex:8];
+        cosmosdbOperationCallback(nil, [self generateResponseWithStatusCode:MSHTTPCodesNo400BadRequest], nil);
+      });
+
+  // When
+  [self.sut performCosmosDbOperationWithPartition:kMSPartitionTest
+                                       documentId:kMSDocumentIdTest
+                                       httpMethod:kMSHttpMethodPost
+                                             body:nil
+                                additionalHeaders:nil
+                                completionHandler:^(NSData *_Nullable responseBody, NSHTTPURLResponse *_Nullable response,
+                                                    NSError *_Nullable error) {
+                                  XCTAssertNil(responseBody);
+                                  actualResponse = response;
+                                  actualError = error;
+                                  [expectation fulfill];
+                                }];
+
+  // Then
+  [self waitForExpectationsWithTimeout:100
```

Comment:
Maybe use less than 100 here so the tests does not hang too long when it breaks 😬 ?

Reply:
Oops.. That was for my test. 😓